torchgeo/datasets/geo.py

```diff
@@ -392,7 +400,7 @@ def __init__(
         i = 0
         filename_regex = re.compile(self.filename_regex, re.VERBOSE)
         for filepath in self.files:
-            match = re.match(filename_regex, os.path.basename(filepath))
+            match = re.match(filename_regex, os.path.basename(str(filepath)))
```

Comment:
I don't think this needs to be cast to a str, right?

Reply:
This will need to be ``str`` as join does not support concatenation of ``bytes`` and ``str``.

Reply:
Possibly ``bytes`` and ``os.PathLike`` is not supported in ``re.match``. Either both arguments in ``re.match`` should be ``str`` or ``bytes``.
[More info here](https://docs.python.org/3/library/re.html)